jackson-modules/jackson-conversions-2/src/test/java/com/baeldung/jackson/booleanAsInt/BooleanAsIntegerUnitTest.java

```diff
@@ -0,0 +1,132 @@
+package com.baeldung.jackson.booleanAsInt;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonFormat.Shape;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class BooleanAsIntegerUnitTest {
+
+    private ObjectMapper mapper;
+
+    @BeforeEach
+    public void setup() {
+        mapper = new ObjectMapper();
+    }
+
+    @Test
+    public void givenBoolean_serializedAsInteger() throws Exception {
+        GameAnnotatedByJsonFormat
+          game = new GameAnnotatedByJsonFormat(1L, "My Game");
+        game.setPaused(true);
+        game.setOver(false);
+        String json = mapper.writeValueAsString(game);
+
+        assertThat(json)
+          .isEqualTo("{\"id\":1,\"name\":\"My Game\",\"paused\":1,\"over\":0}");
+    }
+
+    @Test
+    public void givenInteger_deserializedAsBooleanByDefault() throws Exception {
+        // Integer "1" and "0" values deserialized correctly out of the box.
+        // No configuration or @JsonFormat annotation needed.
+        String json = "{\"id\":1,\"name\":\"My Game\",\"paused\":1,\"over\":0}";
+        Game game = mapper.readValue(json, Game.class);
+
+        assertThat(game.isPaused()).isEqualTo(true);
```

Comment:
You could simply use the standard assertTrue here and assertFalse on the line below.